core/tests/test_apps.eos.check_action.py

```diff
@@ -2,63 +2,96 @@
+        EosActionBuyRam,
+        EosActionBuyRamBytes,
+        EosActionDelegate,
+        EosActionDeleteAuth,
+        EosActionLinkAuth,
+        EosActionNewAccount,
+        EosActionRefund,
+        EosActionSellRam,
+        EosActionTransfer,
+        EosActionUnlinkAuth,
+        EosActionUpdateAuth,
+        EosActionVoteProducer,
+        EosAsset,
+        EosActionCommon,
+    )
 
+# Way of easily filling the required arguments into protobuf messages
+common = EosActionCommon(
+    account=5,
+    name=5,
+)
+kwargs = {
+    "payer": 5,
+    "receiver": 5,
+    "account": 5,
+    "bytes": 5,
+    "sender": 5,
+    "permission": 5,
+    "owner": 5,
+    "voter": 5,
+    "proxy": 5,
+    "parent": 5,
+    "auth": 5,
+    "code": 5,
+    "type": 5,
+    "requirement": 5,
+    "creator": 5,
+    "name": 5,
+    "active": 5,
+    "sender": 5,
+    "memo": 5,
+    "net_quantity": 5,
+    "cpu_quantity": 5,
+    "transfer": 5,
+    "quantity": 5,
+}
 
 @unittest.skipUnless(not utils.BITCOIN_ONLY, "altcoin")
 class TestEosActions(unittest.TestCase):
     def test_check_action(self):
         # return True
-        self.assertEqual(check_action(EosTxActionAck(buy_ram=EosActionBuyRam()), 'buyram', 'eosio'), True)
-        self.assertEqual(check_action(EosTxActionAck(buy_ram_bytes=EosActionBuyRamBytes()), 'buyrambytes', 'eosio'), True)
-        self.assertEqual(check_action(EosTxActionAck(sell_ram=EosActionSellRam()), 'sellram', 'eosio'), True)
-        self.assertEqual(check_action(EosTxActionAck(delegate=EosActionDelegate()), 'delegatebw', 'eosio'), True)
-        self.assertEqual(check_action(EosTxActionAck(undelegate=EosActionDeleteAuth()), 'undelegatebw', 'eosio'), True)
-        self.assertEqual(check_action(EosTxActionAck(refund=EosActionRefund()), 'refund', 'eosio'), True)
-        self.assertEqual(check_action(EosTxActionAck(vote_producer=EosActionVoteProducer()), 'voteproducer', 'eosio'), True)
-        self.assertEqual(check_action(EosTxActionAck(update_auth=EosActionUpdateAuth()), 'updateauth', 'eosio'), True)
-        self.assertEqual(check_action(EosTxActionAck(delete_auth=EosActionDeleteAuth()), 'deleteauth', 'eosio'), True)
-        self.assertEqual(check_action(EosTxActionAck(link_auth=EosActionLinkAuth()), 'linkauth', 'eosio'), True)
-        self.assertEqual(check_action(EosTxActionAck(unlink_auth=EosActionUnlinkAuth()), 'unlinkauth', 'eosio'), True)
-        self.assertEqual(check_action(EosTxActionAck(new_account=EosActionNewAccount()), 'newaccount', 'eosio'), True)
-        self.assertEqual(check_action(EosTxActionAck(transfer=EosActionTransfer()), 'transfer', 'not_eosio'), True)
-        self.assertEqual(check_action(EosTxActionAck(unknown=[]), 'unknown', 'not_eosio'), True)
-        self.assertEqual(check_action(EosTxActionAck(unknown=[]), 'buyram', 'buygoods'), True)
+        self.assertEqual(check_action(EosTxActionAck(common=common, buy_ram=EosActionBuyRam(**kwargs)), 'buyram', 'eosio'), True)
```

Comment:
there's only `not None` check in `check_action`, right? you should be able to get away with `buy_ram=object()` and dtto for other cases

Reply:
Good idea! Done in 329a038